backend/src/routes/helpers/validation.ts

```diff
@@ -10,11 +11,15 @@ type RequestSchema<Output, TD extends z.ZodTypeDef> = z.Schema<
 
 function validateRequest<Output, TD extends z.ZodTypeDef, Input>(
   schema: RequestSchema<Output, TD>,
-  handler: (parsedReq: Output, res: Response, rawReq: Request) => unknown
+  handler: (
+    parsedReq: Output,
+    res: Response,
+    rawReq: CallHomeRequest
+  ) => unknown
 ) {
-  return async (req: Request, res: Response) => {
+  return async (req: CallHomeRequest, res: Response) => {
     try {
-      const parsedReq = schema.parse(req as Partial<Request>);
+      const parsedReq = schema.parse(req as Partial<CallHomeRequest>);
```

Comment:
`schema.parse` will remove types not in the schema. Hence, `parsedReq` will only contain the stuff defined in the schema.

The `req` in the 3rd return value will need to be used if stuff like `req.log.error` is to be used in the router code.

Reply:
Yup. That was the intent.